gdal/swig/python/osgeo/gdal.py

```diff
@@ -976,7 +976,7 @@ def RasterizeOptions(options = [], format = None,
          xRes = None, yRes = None, targetAlignedPixels = False,
          bands = None, inverse = False, allTouched = False,
          burnValues = None, attribute = None, useZ = False, layers = None,
-         SQLStatement = None, SQLDialect = None, where = None,
+         SQLStatement = None, SQLDialect = None, where = None, optim = None,
```

Comment:
swig/python/osgeo/gdal.py is a generated file. You should also make the same change in swig/include/python/gdal_python.i which is its source
(for conveniency both this SWIG generated file and the swig/include/python/gdal_python.i file are kept in revision control. Currently the generated file is regenerated with SWIG 3.0.8)